plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFEOutput.sol

```diff
@@ -0,0 +1,168 @@
+
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(args.outputUtxoPos);
+        uint16 outputIndex = UtxoPosLib.outputIndex(utxoPos);
+        require(
+            PaymentInFlightExitModelUtils.isOutputPiggybacked(ife, outputIndex),
+            "Output is not piggybacked"
+        );
+
+        verifyInFlightTransactionIncludedInPlasma(controller, args);
+        verifyChallengingAndInFlightTransactionAreDifferent(args);
+        verifyOutputType(controller, args);
+        verifyChallengingTransactionSpendsOutput(controller, args, ife);
+
+        PaymentInFlightExitModelUtils.cancelOutputPiggyback(ife, outputIndex);
+
+        //pay bond to challenger
+        msg.sender.transfer(PIGGYBACK_BOND);
+        emit InFlightExitOutputBlocked(msg.sender, keccak256(args.inFlightTx), outputIndex);
+    }
+
+    function verifyInFlightTransactionIncludedInPlasma(
+        Controller memory controller,
+        PaymentInFlightExitRouterArgs.ChallengeOutputSpent memory args
+    )
+        private
+        view
+    {
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(args.outputUtxoPos);
+        TxFinalization.Verifier memory finalizationVerifier = TxFinalization.moreVpVerifier(
+            controller.framework,
+            args.inFlightTx,
+            utxoPos.txPos(),
+            args.inFlightTxInclusionProof
+        );
+
+        require(TxFinalization.isStandardFinalized(finalizationVerifier), "In-flight transaction not finalized");
+    }
+
+    function verifyChallengingAndInFlightTransactionAreDifferent(
+         PaymentInFlightExitRouterArgs.ChallengeOutputSpent memory args
+    )
+        private
+        pure
+    {
+        bytes32 ifeHash = keccak256(args.inFlightTx);
+        bytes32 challengingTxHash = keccak256(args.challengingTx);
+        require(ifeHash != challengingTxHash, "In-flight transaction and challenging transaction must be different");
+    }
+
+    function verifyOutputType(
+        Controller memory controller,
+        PaymentInFlightExitRouterArgs.ChallengeOutputSpent memory args
+    )
+        private
+        view
+    {
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(args.outputUtxoPos);
+        uint16 outputIndex = UtxoPosLib.outputIndex(utxoPos);
+        WireTransaction.Output memory output = WireTransaction.getOutput(args.inFlightTx, outputIndex);
+        OutputGuardModel.Data memory outputGuardData = OutputGuardModel.Data({
+            guard: output.outputGuard,
+            outputType: args.outputType,
+            preimage: args.outputGuardPreimage
+        });
+        IOutputGuardHandler handler = controller.outputGuardHandlerRegistry
+                                                .outputGuardHandlers(args.outputType);
+
+        require(address(handler) != address(0),
+            "Does not have outputGuardHandler registered for the output type");
+
+        require(handler.isValid(outputGuardData),
```

Comment:
This id done to verify if challenger provided the right output type.
I think we need that to make sure challenger does not pick a spending condition that would "pass" and prevent a legitimate exit.

Reply:
I had the same issue on my process IFE PR, as a result, I introduced some extra data fields on the PaymentExitModel for bonds. Probably we can sync to do the same way?

Reply:
Ok. You mean saving that as a field in PaymentExitDataModel.InFlightExit struct?

Reply:
Cool, I'll do it with other changes that come up during the cr.